### ISSUE TITLE
Question on Reward - printing during internal inference

### ISSUE THREAD

**Issue by Guidosalimbeni** [Unity-Technologies/ml-agents#1067]
Hello,
I am enjoying very much your plugin for Unity3D. It is great.
I have a question about rewards. 
Is there a way to Debug.Log (print in the console in Unity) the reward values while the agent is in inference mode using a trained ML model (internal brain)?
More important question: does the agent need the reward logic to be present in Unity3D after the agent has been already trained. I am asking this cause I found a way to train the agent using ML-agents but by opening another 'socket' for the reward's logic. By intuition, I thought that once the agent is trained, and the weights are set in the imported and trained model (Neural Network) the agent doesn't need the rewards to be set for each action during the inference. But I am not sure.
Since I have this doubt, it would be very helpful for me to print on the console the reward of the agent during inference, but I don't understand how I can do this?
The fact is that (because the way I implemented the reward logic out of unity) I cannot 'see' the agent reaching a specific goal, which can be visually evaluated (such as in your example .. the 3Dball staying on the platform).. so I need to print somehow the reward during inference.
Please let me know if you need more details.
Best regards
Guido

**Comment by unityjeffrey** [comment]
hi @Guidosalimbeni thank you for reaching out.  You should be able to collect the reward of the agent when its set to an internal brain using the bytes file.  One approach would be to write a script that is a component of an agent.  From this script, you could reference the agent and call the GetReward() function after each FixedUpdate.  And from this, you could log this to the console.  Let me know if this approach would satisfy what you are looking to do.

**Comment by Guidosalimbeni** [comment]
Thank you!
I really would like to understand this point better, if it is possible.
I set the reward logic coming from an external python script via a socket. I trained the agent to move to the left of the screen and it is working as aspected. What I understand is that the reward logic is needed to train the agent so that the agent can map state observations to actions. Once the agent is trained, and the agent is tested in inference mode, there is no need of a working reward logic anymore. Am I correct?

To test this, I plugged the code as you suggested in Unity (not sure it is correct though) and the reward is not updating but the agent is behaving correctly. Since the external python script is not running during inference it doesn't surprise me that the GetReward() returns 0 all the time. But the agent is moving (slowly) to the left as aspected.

So.. it is true that I don't need a working reward logic (reward assignment) during inference? Or am I missing something?


public class RewardGet : MonoBehaviour {


```
    AgSimpleCube agent;

    // Use this for initialization
    void Start () {

        agent = this.GetComponent<AgSimpleCube>();
		
	}
	
	// Update is called once per frame
	void Update () {
        //Debug.Log("ciao");
        Debug.Log(agent.GetReward());

    }
}
```
